src/app/services/transmart-resource/transmart-mapper.ts

```diff
@@ -236,6 +238,103 @@ export class TransmartMapper {
     return elements;
   }
 
+  public static mapStudiesDimensions(transmartStudies: TransmartStudy[]) {
+    const highDims = ['assay', 'projection', 'biomarker', 'missing_value', 'sample_type', 'end time']
+    let studiesDimensions = new TransmartStudyDimensions();
+
+    if (transmartStudies && transmartStudies.length > 0) {
+      let dimensions = new Array<Dimension>();
+
+      // get default table format for the study
+      if (transmartStudies.length === 1 && transmartStudies[0].metadata != null
+        && transmartStudies[0].metadata.defaultTabularRepresentation != null) {
+        let rowDimensions = new Array<string>();
+        let columnDimensions = new Array<string>();
+        transmartStudies[0].metadata.defaultTabularRepresentation.columnDimensions.forEach((dimName: string) =>
+          columnDimensions.push(dimName));
+        transmartStudies[0].metadata.defaultTabularRepresentation.rowDimensions.forEach((dimName: string) =>
+          rowDimensions.push(dimName));
+
+        studiesDimensions.defaultTableRepresentation = new TransmartTableState(rowDimensions, columnDimensions);
+      }
+
+      // get dimension arrays for each study
+      let availableDimensions = transmartStudies.map(study => study.dimensions);
+
+      // sort to get the shortest dimension at the beginning of the array
+      availableDimensions.sort(function (a, b) {
+        return a.length - b.length;
+      });
+
+      // get common dimensions for all the studies
+      let commonDimensions = availableDimensions.shift().filter(function (v) {
+        return availableDimensions.every(function (a) {
+          return a.indexOf(v) !== -1;
+        });
+      });
+
+      commonDimensions.forEach((name: string) => {
+        if (highDims.indexOf(name) === -1) {
+          dimensions.push(new Dimension(name));
+        }
+      });
+      dimensions.forEach((dimension: Dimension) => studiesDimensions.availableDimensions.push(dimension));
+    }
+
+    return studiesDimensions;
+  }
+
+  public static mapDefaultDimensionsRepresentation(studiesDimensions: TransmartStudyDimensions, dataTable: DataTable) {
```

Comment:
normally if we create a mapper function, it maps something to something else, so maybe it should just take in the transmart study dims, and output the row/col dims that data table can use?